pkg/plugins/common/postgres/pq_listener.go

```diff
@@ -32,22 +32,24 @@ func (p *pqListener) Close() error {
 	return p.listener.Close()
 }
 
-func NewListener(cfg config.PostgresStoreConfig, log logr.Logger) (Listener, error) {
+func NewListener(cfg config.PostgresStoreConfig, log logr.Logger, stop <-chan struct{}) (Listener, error) {
```

Comment:
I don't think we need to pass the `stop` channel here. `pqListener` already has the method `Close` and I think it's the only method clients should use to terminate `pqListener`. We can create a `stop` channel inside `NewListener`, put it into the `pqListener` field, and then call `close(p.stop)` from the `Close` method